From a5bcb18b42cfc0485397da9f48be6a73de86dfac Mon Sep 17 00:00:00 2001
From: Max Krummenacher <max.krummenacher@toradex.com>
Date: Sat, 10 May 2025 14:21:35 +0200
Subject: [PATCH] video: fbdev: mxc: make uapi ipu.h, pxp_dma.h compatible with
 c23 standard

In C23 standard bool, false, true are reserved keywords.
Make ipu.h, pxp_dma.h uapi include file compatible with C23 standard.

Upstream-Status: Pending
Signed-off-by: Max Krummenacher <max.krummenacher@toradex.com>
---
 include/uapi/linux/ipu.h     | 4 ++++
 include/uapi/linux/pxp_dma.h | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/include/uapi/linux/ipu.h b/include/uapi/linux/ipu.h
index c92f292bcc9d..a7b60730bbd4 100644
--- a/include/uapi/linux/ipu.h
+++ b/include/uapi/linux/ipu.h
@@ -29,9 +29,13 @@
 #include <linux/videodev2.h>
 
 #ifndef __KERNEL__
+#if defined __STDC_VERSION__ && __STDC_VERSION__ > 201710L
+/* bool, true and false are keywords.  */
+#else
 #ifndef __cplusplus
 typedef unsigned char bool;
 #endif
+#endif
 #define irqreturn_t int
 #define dma_addr_t int
 #define uint32_t unsigned int
diff --git a/include/uapi/linux/pxp_dma.h b/include/uapi/linux/pxp_dma.h
index 9b5228504095..e2deff07516d 100644
--- a/include/uapi/linux/pxp_dma.h
+++ b/include/uapi/linux/pxp_dma.h
@@ -25,8 +25,12 @@
 
 #ifndef __KERNEL__
 typedef unsigned long dma_addr_t;
+#if defined __STDC_VERSION__ && __STDC_VERSION__ > 201710L
+/* bool, true and false are keywords.  */
+#else
 typedef unsigned char bool;
 #endif
+#endif
 
 /*  PXP Pixel format definitions */
 /*  Four-character-code (FOURCC) */
-- 
2.42.0

